library/src/androidTest/java/org/xmtp/android/library/InstrumentedTest.kt

```diff
@@ -97,4 +100,65 @@ class InstrumentedTest {
         assert(contact!!.v2.keyBundle.identityKey.hasSignature())
         assert(contact.v2.keyBundle.preKey.hasSignature())
     }
+
+    @Test
+    fun testCanPaginateV2Messages() {
+        val bob = PrivateKeyBuilder()
+        val alice = PrivateKeyBuilder()
+        val clientOptions =
+            ClientOptions(api = ClientOptions.Api(env = XMTPEnvironment.LOCAL, isSecure = false))
+        val bobClient = Client().create(bob, clientOptions)
+        // Publish alice's contact
+        Client().create(account = alice, clientOptions)
+        val convo = bobClient.conversations.newConversation(
+            alice.address,
+            context = InvitationV1ContextBuilder.buildFromConversation("hi")
+        )
+        // Say this message is sent in the past
+        val date = Date()
+        date.time = date.time - 10000
+        convo.send(text = "10 seconds ago", sentAt = date)
```

Comment:
Kotlin's Date should have `minus()` and `plus()` methods to make this a little easier! https://kotlinlang.org/api/kotlinx-datetime/kotlinx-datetime/kotlinx.datetime/minus.html

Reply:
Ya I looked into this but you have to change the date to a instant and then back to a date which felt like a lot for a test.